envoy/stream_info/stream_info.h

```diff
@@ -721,6 +721,15 @@ class StreamInfo {
     downstream_info.setUpstreamBytesMeter(upstream_info.getUpstreamBytesMeter());
     upstream_info.setDownstreamBytesMeter(downstream_info.getDownstreamBytesMeter());
   }
+
+  /**
+   * Filter State object to be shared between upstream and downstream filters.
```

Comment:
I think if we end up going this route we're going to want to be awfully clear that this is downstream L7 filters and upstream L4 filters.

I also wonder if instead of adding another set of accessors, if there would be a way to tag information that's added as "information that should be communicated to upstream stream info" WDYT?

Reply:
Yes, this is more confusing than helpful. I think an explicit export is probably better (either in the filter state itself or in the transport socket).